doc/faq/readme.md

```diff
@@ -43,3 +43,111 @@ You will see this issue only if serial upload was not followed by a physical res
 
 Ref. [#1017](https://github.com/esp8266/Arduino/issues/1017), [#1107](https://github.com/esp8266/Arduino/issues/1107), [#1782](https://github.com/esp8266/Arduino/issues/1782)
 
+
+### I'm getting these Watchdog Resets "wdt reset"�How can I avoid them?
+The esp8266 is equipped with two watchdog timers: A software watchdog and a hardware watchdog. If you don't understand why microcontrollers need a watchdog (timer), you may first read [here](http://www.embedded.com/electronics-blogs/beginner-s-corner/4023849/Introduction-to-Watchdog-Timers).
+
+When your sketch is running�either in the `setup` or during the `loop`�you have to make sure to feed the watchdog periodically. Otherwise "it will bite" and the watchdog will restart your sketch. The basic operation is illustrated in the following picture:
+![wdt basic operation](pictures/wdt_basic_operation.png)
+
+
+**How does the esp8266 Arduino Core feed the watchdog(s)?**
+From the perspective of your Arduino sketch, the basic feeding is implicit: Everytime an iteration of your main loop starts again, i.e. it hits the `loop` statement both watchdogs are fed; cf. (1) in the figure above. Esp8266 Arduino libraries, especially core libraries like for instance the `ESP8266WiFi` library take care of feeding the watchdogs, too. Thus, normally you don't have to feed them explicitly.
+
+**When and why should I feed the watchdog(s) explicitly?**
+The software watchdog bites exactly every 3.2 seconds, the hardware watchdog every 7�8 seconds. If your sketch contains code which could exceed running for 3.2 seconds you must feed the watchdogs. I.e. you have to feed them explicitly by calling (cf. (1) in the figure above):
```

Comment:
It is probably worth noting that feeding the watchdog is not the only problem if some code runs for more that a second straight. WiFi and TCP/IP stack will not work as expected after this: station mode wifi will disconnect due to beacon timeout, and TCP stack may get flooded with packets.